### PR TITLE
feat: update to DAPI Client 0.14 and refactor transport layer

### PR DESCRIPTION
**Pull request by shumkov**

### CODE REVIEW

File: src/types/Wallet/Wallet.js [hunk 1]
```diff
@@ -110,11 +113,23 @@ class Wallet {
         this.storage.importAddresses(opts.cache.addresses, this.walletId);
       }
     }
-    if (this.offlineMode) {
-      this.transporter = { isValid: false };
-    } else {
-      this.transporter = transporters.resolve(opts.transporter);
+
+    if (!this.offlineMode) {
+      if (opts.transport && opts.transport.network) {
```
Comment (Alex-Werner) [inline src/types/Wallet/Wallet.js:118]: Why removing validation of a transport ?

File: src/types/Wallet/Wallet.js [hunk 2]
```diff
@@ -110,11 +113,23 @@ class Wallet {
         this.storage.importAddresses(opts.cache.addresses, this.walletId);
       }
     }
-    if (this.offlineMode) {
-      this.transporter = { isValid: false };
```
[the remark this answers is not in the document]
Comment (shumkov) [reply src/types/Wallet/Wallet.js:114]: Validation adds complexity to the code and we don’t need it if we implement Transport interface and use typings.

File: src/transport/DAPIClientTransport/methods/getBestBlockHash.spec.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    transport = new DAPIClientTransport(clientMock);
+  })
+
+  afterEach(() => {
+    transport.disconnect();
+  })
+
+  it('should works', async () => {
```
Comment (antouhou) [inline src/transport/DAPIClientTransport/methods/getBestBlockHash.spec.js:26]: works -> work

File: src/transport/DAPIClientTransport/methods/getBlockHeaderByHeight.spec.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    transport = new DAPIClientTransport(clientMock);
+  })
+
+  afterEach(() => {
+    transport.disconnect();
+  })
+
+  it('should works', async () => {
```
Comment (antouhou) [inline src/transport/DAPIClientTransport/methods/getBlockHeaderByHeight.spec.js:27]: "work"

File: src/transport/DAPIClientTransport/methods/getStatus.spec.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    transport = new DAPIClientTransport(clientMock);
+  })
+
+  afterEach(() => {
+    transport.disconnect();
+  })
+
+  it('should works', async () => {
```
Comment (antouhou) [inline src/transport/DAPIClientTransport/methods/getStatus.spec.js:28]: Well it's virtually in every test

File: src/types/Wallet/Wallet.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (antouhou) [reply src/types/Wallet/Wallet.js:114]: I would agree with that. Code shouldn't be checking if the implemenation of the interface is correct or not

[the remark this answers is not in the document]
Comment (Alex-Werner) [reply src/types/Wallet/Wallet.js:114]: Make sense, closing.

**Pull request merged**